pkg/scheduler/framework/plugins/legacy_registry.go

```diff
@@ -428,32 +430,95 @@ func NewLegacyRegistry() *LegacyRegistry {
 				pluginConfig = append(pluginConfig,
 					config.PluginConfig{Name: serviceaffinity.Name, Args: args.ServiceAffinityArgs})
 			}
-			return
+			return pluginConfig
 		})
 	registry.registerPriorityConfigProducer(EvenPodsSpreadPriority,
-		func(args ConfigProducerArgs) (plugins config.Plugins, pluginConfig []config.PluginConfig) {
+		func(args ConfigProducerArgs, plugins *config.Plugins, pluginConfig []config.PluginConfig) []config.PluginConfig {
 			plugins.PreScore = appendToPluginSet(plugins.PreScore, podtopologyspread.Name, nil)
 			plugins.Score = appendToPluginSet(plugins.Score, podtopologyspread.Name, &args.Weight)
-			return
+			return pluginConfig
 		})
 
 	return registry
 }
 
+// GetPredicateConfigs returns predicates configuration: ones that will run as fitPredicates and ones that will run
+// as framework plugins. Specifically, a predicate will run as a framework plugin if a plugin config producer was
+// registered for that predicate.
+// Note that the framework executes plugins according to their order in the Plugins list, and so predicates run as plugins
+// are added to the Plugins list according to the order specified in predicates.Ordering().
+func (lr *LegacyRegistry) GetPredicateConfigs(keys sets.String, args *ConfigProducerArgs, plugins *config.Plugins, pluginConfig []config.PluginConfig) ([]config.PluginConfig, error) {
```

Comment:
should we return both Plugins and PluginConfig and call the function AppendPredicatePlugins?

Reply:
Done